mod/blog/actions/blog/save.php

```diff
@@ -91,7 +91,8 @@
 		case 'container_guid':
 			// this can't be empty or saving the base entity fails
 			if (!empty($value)) {
-				if (can_write_to_container($user->getGUID(), $value)) {
+				$container = get_entity($value);
+				if ($container && $container->canWriteToContainer()) {
```

Comment:
Could we start requesting subtype-specific permissions when we can? 
`$container->canWriteToContainer(0, 'object', 'blog')`